src/usr/local/www/vpn_openvpn_server.php

```diff
@@ -602,6 +606,10 @@
 			}
 			$server['caref'] = $pconfig['caref'];
 			$server['crlref'] = $pconfig['crlref'];
+			if ($pconfig['ocspcheck']) {
+				$server['ocspcheck'] = "yes";
+			}
+			$server['ocspurl'] = $pconfig['ocspurl'];
```

Comment:
The URL must be validated before it is saved, especially since it is passed to a shell command. There is a function for this already in `util.inc`, called `is_URL($url)`, which should work.

Reply:
but it's already checked with Form/Input placeholder type 'url'
do we need another check?

Reply:
Yes, that check is only client-side in JavaScript. It must also be checked when submitted, since malicious clients can submit things other ways, or someone may have JS disabled.

Reply:
ok, done